libdnf5/rpm/package_sack_impl.hpp

```diff
@@ -161,6 +164,9 @@ class PackageSack::Impl {
     int cached_solvables_size{0};
     PackageId running_kernel;
 
+    // ids of packages created from local rpm files
+    std::unordered_set<Id> local_rpm_ids;
```

Comment:
I am not really sure about this.
In my mind the `Repo::Type::AVAILABLE` repos are not meant for locally added rpms. I would be interested in an opinion from @jrohel.

On the other hand I understand that our options are limited here.

Given that we are interested only in the repo ids would it be possible to perhaps not load the repo definitions at all and create only `Repo::Type::COMMANDLINE` repos with the specified id when needed by the `add_stored_transaction_package(...)`?
This would have the advantage that package replays with repo ids that are no longer defined on the system would still have the original repoid instead of `@stored_transaction`.

Reply:
Yes, this is also possibility. Let me experiment with this approach a bit...

Reply:
I've created an alternative PR which does not create repositories from the system configuration and creates `COMMANDLINE` type repos on the fly as they are needed: https://github.com/rpm-software-management/dnf5/pull/2093